### PR TITLE
Add affinity in prometheus CR to avoid spot instances

### PR DESCRIPTION
**Pull request by marieroque**
towards https://github.com/giantswarm/giantswarm/issues/28239

Suggestion from here https://github.com/giantswarm/giantswarm/issues/28447#issuecomment-1764638339

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Test on Workload cluster.

### CODE REVIEW

**Comment by marieroque** [comment]
Tested on `anteater/deu01`

**Comment by marieroque** [comment]
Tested on `gaia` MC ✅ 
Tested on `gaia` WC ✅

**Review by QuantumEnigmaa — APPROVED** [review]
LGTM but I guess it should be better to wait for other team members to approve as well.

File: helm/prometheus-agent/charts/prometheus-agent/values.yaml [hunk 1]
```diff
@@ -8,6 +8,28 @@ image:
   sha: ""
   pullPolicy: "Always"
 
+affinity:
+  nodeAffinity:
+    preferredDuringSchedulingIgnoredDuringExecution:
```
Comment (QuentinBisson) [inline helm/prometheus-agent/charts/prometheus-agent/values.yaml:13]: Why is this not indented ? Also, i'd probably only have affinity like Loki or tempo do

**Comment by QuentinBisson** [comment]
Tested on Gaia

**Pull request merged**